package.json

```diff
@@ -4,7 +4,10 @@
   "scripts": {
     "build": "npm run build:widget && npm run build:app",
     "build:app": "vite build",
-    "build:widget": "rollup -c",
+    "build:dev:widget": "rollup -c",
+    "build:dev:svelte-package": "svelte-package",
+    "build:dev:types": "rollup -c ./rollup.types.config.js",
+    "build:widget": "npm-run-all build:dev:*",
```

Comment:
I see you've added some additional commands here, but what's your recommendation for adding them into the `npm run dev` command? The issue is this command currently doesn't build any types, so when using the e2e apps for local dev verification, you end up with a bunch of type errors. One then has to run these separate commands to fix the errors for each iteration of develop-test-develop cycle. It's not a huge deal, but it just adds up if making significant changes over time.

It would be nice if the types just built with everything else with the one `npm run dev`. If this is too process intensive, is there a different way you would recommend to reference types from the e2e apps to both avoid errors and not have to run separate commands?

Reply:
`build:widget` gets called by `dev` ... right?

Reply:
Rereading this i think maybe the npm-run-all command is confusing here - it basically runs all of the above commands (i believe in the order they are listed in the file) so any command that matches this pattern

`build:dev:*` so thats how thats working now

Reply:
`npm run build:widget` only gets called initially, but there's no "watch" command, so one would have to stop and restart the `npm run dev` command. `npm run dev:widget` would be the command that would be best to handle the generation of types. Please test this locally and ensure this isn't too process intensive though. We don't want to slow the dev command to a crawl.

Reply:
so is the ask to build the types on every save? If so - just so we understand what we are doing

We save a file:
we all rollup once for the widget itself, source code, one call to svelte-package to save the types, then another call to rollup to bundle the types, and then the app would need to get reloaded with vite.

Is that what we want? I can't imagine this dev flow being sustainable?

Reply:
i have to think about this, i wonder if docker is the best way to develop in this case? really not sure

Reply:
Being that this is so resource intensive, I think keeping it out of the "watched" changes is okay.